src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -89,15 +93,20 @@ private static boolean isValidPath(Path filePath) {
     /**
      * Saves all data to this storage file.
      *
-     * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws StorageOperationException
+     *             if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
 
-        /* Note: Note the 'try with resource' statement below.
-         * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
+        /*
+         * Note: Note the 'try with resource' statement below. More info:
+         * https://docs.oracle.com/javase/tutorial/essential/exceptions/
```

Comment:
The exception should be thrown from StorageFile::save() method.
The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.